Formula/nifti_clib.rb

```diff
@@ -0,0 +1,42 @@
+class NiftiClib < Formula
+  desc "C libraries for NIFTI support"
+  homepage "https://github.com/NIFTI-Imaging/nifti_clib"
+  url "https://github.com/leej3/nifti_clib/archive/v3.0.0.tar.gz"
+  sha256 "fe6cb1076974df01844f3f4dab1aa844953b3bc1d679126c652975158573d03d"
+
+  depends_on "cmake" => :build
+  depends_on "help2man" => :build
+  depends_on "make" => :build
+  uses_from_macos "zlib"
+
+  def install
+    # Install static libraries
+    system "cmake", ".",
+      "-DCMAKE_BUILD_TYPE=RELWITHDEBINFO",
+      "-DCMAKE_INSTALL_PREFIX=#{prefix}",
+      "-DNIFTI_INSTALL_MAN_DIR=#{man}",
+      "-DDOWNLOAD_TEST_DATA=OFF",
+      "-DTEST_INSTALL=OFF"
+    system "make"
+    system "cmake",
+      "-DCOMPONENT=RuntimeLibraries",
+      "-P", "cmake_install.cmake",
+      "install"
+
+    # Install shared object libraries, and all other components of build
+    system "cmake", ".",
+      "-DBUILD_SHARED_LIBS=ON"
+    system "make", "install"
+
+    # Tests need to be run in build tree:
+    system "ctest", "-LE", "NEEDS_DATA"
+  end
+
+  test do
+    # The installed folder is not in the path, so use the entire path to any
+    # executables being tested: `system "#{bin}/program", "do", "something"`.
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thanks for the feedback. The good news is I have added a better test. The bad news is that I found a bug (or perhaps that is also good news).

The cmake module exporting targets contains hardcoded paths. I'll have a look around for how to fix that. I'll patch it hear and then upstream a fix if appropriate. That sound good?

Reply:
> The cmake module exporting targets contains hardcoded paths. I'll have a look around for how to fix that. I'll patch it hear and then upstream a fix if appropriate. That sound good?

Mostly yes, but per our policy https://docs.brew.sh/Acceptable-Formulae#niche-or-self-submitted-stuff we won't accept the new formula until it builds without patches, so definitely submit them upstream.